pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -124,22 +124,49 @@ public void setLogWriter(PrintWriter printWriter) {
    * Gets the name of the host the PostgreSQL database is running on.
    *
    * @return name of the host the PostgreSQL database is running on
+   * @deprecated use {@link #getServerNames()}
    */
+  @Deprecated
   public String getServerName() {
-    return serverName;
+    return serverNames[0];
+  }
+
+  /**
+   * Gets the name of the host(s) the PostgreSQL database is running on.
+   *
+   * @return name of the host(s) the PostgreSQL database is running on
+   */
+  public String[] getServerNames() {
+    return serverNames;
   }
 
   /**
    * Sets the name of the host the PostgreSQL database is running on. If this is changed, it will
    * only affect future calls to getConnection. The default value is <tt>localhost</tt>.
    *
    * @param serverName name of the host the PostgreSQL database is running on
+   * @deprecated use {@link #setServerNames(String[])}
    */
+  @Deprecated
   public void setServerName(String serverName) {
     if (serverName == null || serverName.equals("")) {
-      this.serverName = "localhost";
+      this.serverNames = new String[] {"localhost"};
     } else {
-      this.serverName = serverName;
+      this.serverNames = new String[] {serverName};
+    }
+  }
+
+  /**
+   * Sets the name of the host(s) the PostgreSQL database is running on. If this is changed, it will
+   * only affect future calls to getConnection. The default value is <tt>localhost</tt>.
+   *
+   * @param serverNames name of the host(s) the PostgreSQL database is running on
+   */
+  public void setServerNames(String[] serverNames) {
```

Comment:
The logic for inferring empty or null to `"localhost"` should happen for all entries in the array, not just the first. Otherwise you'll get NPEs down the road if called via `setServerNames(new String[] { "foo", null })`. Either validate and throw an Exception at assignment or have the same defaulting logic apply to all entries (seems preferable).